src/traces/heatmap/plot.js

```diff
@@ -337,7 +337,14 @@ function plotOne(gd, plotinfo, cd) {
 
     if(zsmooth) { // best or fast, works fastest with imageData
         var pxIndex = 0,
+            pixels;
+
+        try {
             pixels = new Uint8Array(imageWidth * imageHeight * 4);
```

Comment:
`(new Uint8Array || new Array)(imageWidth * imageHeight * 4)` would fail most of the time in old browsers:
- If you don't include the `typedarray` polyfill, then `Uint8Array` is not just undefined, it's a bad symbol. I suppose we could get around that by saying `window.Uint8Array` but...
- If you *do* include the polyfill, but `imageWidth * imageHeight * 4 > 1e5` then the `Uint8Array` constructor will execute and throw an error.